mos-platform/cpm65/cpm.h

```diff
@@ -95,36 +95,74 @@ extern uint8_t cpm_errno;
 /*  6 */ extern uint8_t cpm_conio(uint8_t b);
```

Comment:
Most of these routines should probably be marked with `__attribute__((leaf))`. This tells the compiler that the underlying assembly routine can't call back into the main program visible at link time via LTO. Otherwise, the compiler would be forced to assume that your assembly routines could do anything, up to including calling `main`. That means it would have to consider any functions on the path from `main` down to these call recursive, and use a cc65-style stack for them.

Marking most/all of these as leaves should allow your C utilities to benefit from llvm-mos's static stack and global zero page allocation capabilities. You only would need to do this for the `_i` variants; anything the compiler can't see through.

Reply:
Done.